kube.libsonnet

```diff
@@ -211,16 +232,32 @@
     assert !self.tty || self.stdin : "tty=true requires stdin=true",
   },
 
+  PodDisruptionBudget(name): $._Object("policy/v1beta1", "PodDisruptionBudget", name) {
+    local this = self,
+    target_pod:: error "target_pod required",
+    spec: {
+      minAvailable: 1,
+      selector: {
+        matchLabels: this.target_pod.metadata.labels,
+      },
+    },
+  },
+
   Pod(name): $._Object("v1", "Pod", name) {
     spec: $.PodSpec,
   },
 
   PodSpec: {
     // The 'first' container is used in various defaults in k8s.
-    default_container:: std.objectFields(self.containers_)[0],
+    local container_names = std.objectFields(self.containers_),
+    default_container:: if std.length(container_names) > 1 then "default" else container_names[0],
     containers_:: {},
 
-    containers: [{ name: $.hyphenate(name) } + self.containers_[name] for name in [self.default_container] + [n for n in std.objectFields(self.containers_) if n != self.default_container]],
+    local container_names_ordered = [self.default_container] + [n for n in container_names if n != self.default_container],
+    containers: [{ name: $.hyphenate(name) } + self.containers_[name] for name in container_names_ordered if self.containers_[name] != null],
+
+    initContainers_:: {},
+    initContainers: [{ name: $.hyphenate(name) } + self.initContainers_[name] for name in std.objectFields(self.initContainers_) if self.initContainers_[name] != null],
```

Comment:
Not particularly a big fan on doing the same here for `initContainers`,
as order for these is "very" relevant, and lost in map key-sorting.

Please add a comment with this caveat, then as `jsonnet` seems to
do alpha-sorting, suggest a workaround if order preserving is needed, e.g.
```
initContainers: { 01_foo: kube.Container(...){...}, 02_bar: kube.Container(...){...} }
```

Reply:
I chose to mention only manipulating initContainers array directly (and not mention sorting via keys), because I feel like we don't have enough tests and process in place to guarantee this order doesn't change at some point.  Eg: I could see passing this through a `std.set()` might even be a feature at some point, and would destroy key-based alpha sorting.  TL;DR: If order is important, I think we *should* use an array and not rely on implicit key-sort.